src/Features/Core/Portable/Snippets/Snippet.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    /// <summary>
+    /// Encapsulates the information that makes up a Snippet.
+    /// What is used to create the CompletionChange in the 
+    /// AbstractSnippetCompletionProvider.
```

Comment:
we shoudl not doc details like this.  For example, snippets might not end up as part of completion at all.